packages/core/src/agent/MessageSender.ts

```diff
@@ -250,7 +252,55 @@ export class MessageSender {
     }
 
     const outboundPackage = await this.packMessage({ message, keys, endpoint: service.serviceEndpoint })
-    await this.outboundTransporter.sendMessage(outboundPackage)
+    outboundPackage.endpoint = service.serviceEndpoint
+    for (const transport of this.outboundTransporters) {
+      if (transport.supportedSchemes.includes(service.protocolScheme)) {
+        await transport.sendMessage(outboundPackage)
+        break
+      }
+    }
+  }
+
+  private async retrieveServicesByConnection(
+    connection: ConnectionRecord,
+    transportPriority?: TransportPriorityOptions
+  ) {
+    this.logger.debug(
+      `Retrieving services for connection '${connection.id}'${' (' && connection.theirLabel && ')'}${
+        (transportPriority?.restrictive && ' restrictively ',
+        transportPriority && ' by priority of the follow schemes: ' + transportPriority.schemes)
+      }`
+    )
```

Comment:
There is a lot happening in this debug statement, I'd prefer to pass it as extra argument to the second parameter

```suggestion
    this.logger.debug(
      `Retrieving services for connection '${connection.id}' (${connection.theirLabel})`,
      {
      	transportPriority
      }
    )
```